gflownet/gflownet.py

```diff
@@ -296,12 +352,14 @@ def sample_actions(
 
     def step(
         self,
-        envs: List,
+        envs: List[GFlowNetEnv],
```

Comment:
You are right, this is how it currently works. The environments should not be too heavy, but it is true that this may be improved. Actually, having this batch class should make this change easier. But it would be a big one anyway, so not now :sweat_smile:.